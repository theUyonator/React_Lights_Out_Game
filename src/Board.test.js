import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Board from "./Board";


describe("<Cell /> rendering", () => {

    describe("Initial board and win state", () => {

        it("renders correctly without crashing", () => {
            render(<Board />)
        });
        
        it("matches snapshot when all lights are on", () => {
            const { asFragment } = render(<Board chanceLightStartsOn={1} />);
            expect(asFragment()).toMatchSnapshot();
        });

        it("matches snapshot when all lights are out", () => {
            const { asFragment } = render(<Board chanceLightStartsOn={0} />);
            expect(asFragment()).toMatchSnapshot();
        });

    })

    // describe("When cells are clicked", () => {

    //     it("says  you won when all lights are turned off", () => {
    //         const { queryByText, getAllByRole }  = render(<Board nrows={3} ncols={3} chanceLightStartsOn={1} />);

    //         // The winning message is not supposed to be in the document as presently rendered 

    //         expect(queryByText("You won the game!")).not.toBeInTheDocument();

    //         // Now we click the middle cell which will toggle all other cells off and we expect 
    //         // the win message to be in the document.

    //         const cells = getAllByRole("button");
    //         fireEvent.click(cells[1]);
    //         expect(queryByText("You won the game!")).toBeInTheDocument();

    //     });

    // })



});