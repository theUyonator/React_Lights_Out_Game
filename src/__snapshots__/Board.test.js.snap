// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> rendering Initial board and win state matches snapshot when all lights are on 1`] = `
<DocumentFragment>
  <table
    class="Board"
  >
    <tbody>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`<Cell /> rendering Initial board and win state matches snapshot when all lights are out 1`] = `
<DocumentFragment>
  <div>
     You won the game! 
  </div>
</DocumentFragment>
`;
